spec/haskell/src/SEL4/Object/SchedContext.lhs

```diff
@@ -723,3 +727,26 @@ This module uses the C preprocessor to select a target architecture.
 >     threadSet (\t -> t { tcbInReleaseQueue = False }) tcbPtr
 >     setReprogramTimer True
 >     return tcbPtr
+
+In preemptible code, the kernel may explicitly mark a preemption point with the "preemptionPoint" function. The preemption will only be taken if an interrupt has occurred and the preemption point has been called "workUnitsLimit" times.
+
+> workUnitsLimit = 0x64
+
+> preemptionPoint :: KernelP ()
```

Comment:
@lsf37 I've (only just) tried translating it automatically. It seems to work fine, except that it'll produce in the `design` spec files `lift` instead of `liftE`. Is this to do with the Haskell translator? The previous version of `preemptionPoint` in the Haskell was just writing a plain `lift`, rather than `liftE` or `withoutPreemption`. What should be done here?

Reply:
Just for the lols, I decided to add

`'lift': 'liftE',`

to the definition of `error_m_map` in the Haskell translator (after line 1900) and it worked. Shall I go that route?

Reply:
This ended up breaking something somewhere else (something about `liftME`). So I've gone for writing `withoutPreemption` instead of `lift` within `preemptionPoint`; this is used in the Haskell definition of other functions which use the `KernelP` monad. Does that suit?